src/zkp/mod.rs

```diff
@@ -26,6 +26,13 @@ use merlin::Transcript;
 use rand::{CryptoRng, RngCore};
 use serde::{de::DeserializeOwned, Serialize};
 
+/// A trait for constructing arbitrary system context.
+///
+/// This context can be turned into bytes and appended to a [`Transcript`].
+pub(crate) trait ProofContext {
+    fn as_bytes(&self) -> Vec<u8>;
+}
+
```

Comment:
Suggestion: You might `impl ProofContext for ()` here since it'll be used all over shortly.